pkg/api/handlers/projects.go

```diff
@@ -370,6 +373,17 @@ func (h *ProjectsHandler) DeleteProject(req api.Context) error {
 		return err
 	}
 
+	if project.Spec.SlackConfiguration != nil {
+		var slackApp v1.OAuthApp
+		if err := req.Get(&slackApp, getSlackOauthAppFromThreadName(project.Name)); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
add a spec.threadname field to oauthapp to create the relationship and then add  to the oauthapps deleterefs an entry so that the oauth app is deleted when the project/thread is deleted